src/test/java/org/openRealmOfStars/starMap/RouteTest.java

```diff
@@ -23,114 +24,210 @@
+    assertEquals("The internal movement speed should be 0.", 0, route.getMy(), 0.1);
+    assertEquals("The distance should be 1.", 1, route.getDistance());
+    assertFalse("The route shouldn't be defending.", route.isDefending());
+    assertFalse("The route shouldn't be fixing.", route.isFixing());
+    assertFalse("The end shouldn't be reached.", route.isEndReached());
+
+    route.makeNextMove();
+
+    assertEquals("The X coordinate should be 5 after the second step.", 5, route.getX());
+    assertEquals("The Y coordinate should be 5 after the second step.", 5, route.getY());
+    assertEquals("The X coordinate should be 5 after the second step.", 5, route.getStartX(), 0.1);
+    assertEquals("The Y coordinate should be 5 after the second step.", 5, route.getStartY(), 0.1);
+    assertEquals("The end X coordinate should be 5.", 5, route.getEndX(), 0.1);
+    assertEquals("The end Y coordinate should be 5.", 5, route.getEndY(), 0.1);
+    assertEquals("The speed should be 1.", 1, route.getFtlSpeed());
+    assertEquals("The internal movement speed should be 0.", 0, route.getMx(), 0.1);
+    assertEquals("The internal movement speed should be 0.", 0, route.getMy(), 0.1);
+    assertEquals("The distance should be 0.", 0, route.getDistance());
+    assertFalse("The route shouldn't be defending.", route.isDefending());
+    assertFalse("The route shouldn't be fixing.", route.isFixing());
+    assertTrue("The end should be reached.", route.isEndReached());
   }
 
   @Test
   public void testIsDefending() {
-    Route route = new Route(3, 3, 5, 5, 0);
-    assertEquals(true,route.isDefending());
-    route = new Route(3, 3, 5, 5, 1);
-    assertEquals(false,route.isDefending());
-    route = new Route(3, 3, 5, 5, Route.ROUTE_FIX);
-    assertEquals(false,route.isDefending());
+    route.setStartX(3);
+    route.setStartY(3);
+    route.setEndX(5);
+    route.setEndY(5);
+    route.setFtlSpeed(Route.ROUTE_DEFEND);
+
+    assertTrue("The route should be defending.", route.isDefending());
+
+    route.setFtlSpeed(1);
+    assertFalse("The route shouldn't be defending when it is attacking.", route.isDefending());
+
+    route.setFtlSpeed(Route.ROUTE_FIX);
+    assertFalse("The route shouldn't be defending when it is fixing.", route.isDefending());
   }
 
   @Test
   public void testIsFixing() {
-    Route route = new Route(3, 3, 5, 5, Route.ROUTE_FIX);
-    assertEquals(true,route.isFixing());
-    route = new Route(3, 3, 5, 5, 1);
-    assertEquals(false,route.isFixing());
-    route = new Route(3, 3, 5, 5, Route.ROUTE_DEFEND);
-    assertEquals(false,route.isFixing());
+    route.setStartX(3);
+    route.setStartY(3);
+    route.setEndX(5);
+    route.setEndY(5);
+    route.setFtlSpeed(Route.ROUTE_FIX);
+
+    assertTrue("The route should be fixing.", route.isFixing());
+
+    route.setFtlSpeed(1);
+    assertFalse("The route shouldn't be fixing when it is attacking.", route.isFixing());
```

Comment:
FTL speed is also used when exploring the. So just saying when "it is moving with FTL".